From 66b3ea10fa94d0f9792d066b9bb6bf9730a0ad8e Mon Sep 17 00:00:00 2001
From: Martin Kampas <martin.kampas@tieto.com>
Date: Tue, 4 Feb 2014 11:54:20 +0100
Subject: Start with --compat switch for sdptool to work

---
 src/bluetooth.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bluetooth.service.in b/src/bluetooth.service.in
index 35e9457..7ac7ff6 100644
--- a/src/bluetooth.service.in
+++ b/src/bluetooth.service.in
@@ -5,7 +5,7 @@ Documentation=man:bluetoothd(8)
 [Service]
 Type=dbus
 BusName=org.bluez
-ExecStart=@libexecdir@/bluetoothd
+ExecStart=@libexecdir@/bluetoothd --compat
 NotifyAccess=main
 #WatchdogSec=10
 #Restart=on-failure
-- 
1.8.5.3

